commons/_shared/bs-lib-components/ui/dev/src/pages/core-components/BsLayoutDefault.vue

```diff
@@ -1,27 +1,61 @@
 <template>
-  <doc-page-core title="BsLayoutDefault" :description="description">
-     <LayoutGalleryGrid class="q-mt-sm"/>
+  <doc-page-core title="BsLayoutDefault">
+    <template #description>
+      The BsLayoutDefault is the default layout for business solutions webapps.<br><br>
+      It is highly customizable, the layout provides 4 child components:<br>
+      <b>BsHeader, BsDrawer, BsContent and BsDocumentation</b> that can be populated with custom vue components.<br>
+      Instead of these components you can use template sections:<br>
+      <code-prism lang="html" :code="componentTemplateAccordancy"></code-prism>
+
+      See the examples in the usage section for more details.
+    </template>
+    <LayoutGalleryGrid class="q-mt-sm"/>
   </doc-page-core>
 </template>
-<script>
-  import DocExample from '../../components/DocExample.vue';
-  import DocPageCore from '../../components/DocPageCore.vue';
-  import LayoutGalleryGrid from "../../components/LayoutGalleryGrid.vue";
+<script lang="ts">
+import DocExample from '../../components/DocExample.vue';
+import DocPageCore from '../../components/DocPageCore.vue';
+import LayoutGalleryGrid from "../../components/LayoutGalleryGrid.vue";
+
+import CodePrism from 'src/components/CodePrism';
+
+import BsHeader from 'app/../src/components/layout/BsHeader.vue';
+import BsLayoutDefault from 'app/../src/components/layout/BsLayoutDefault.vue';
+import BsDocumentation from 'app/../src/components/layout/BsDocumentation.vue';
+import BsDrawer from 'app/../src/components/layout/BsDrawer.vue';
+import BsContent from 'app/../src/components/layout/BsContent.vue';
+
+import { Component, defineComponent } from 'vue';
 
-  export default {
 
+  export default defineComponent({
       components: {
           DocExample, 
           DocPageCore,
           LayoutGalleryGrid,
+          CodePrism,
       },
-      setup() {
-          const description = "The BsLayoutDefault is the default layout for business solutions webapps. It is highly customizable, the layout provides 4 named slots: \
-            header, leftPanel, content and documentation that can be populated with custom vue components. See the examples in the usage section for more details."
-          
-          return {
-              description,
+      computed: {
+        componentToTemplateMapping(): Record<string, any>{
+          const mapping = {} as Record<string, any>;
+
+          function addEntry({ name }: Component, templateName: string) {
+            if (name) mapping[name] = templateName;
           }
-      }
-  }
+
+          addEntry(BsHeader, "header");
+          addEntry(BsDocumentation, "documentation");
+          addEntry(BsDrawer, "leftpanel");
+          addEntry(BsContent, "content");
+
+          return mapping;
+        },
+        componentTemplateAccordancy(): string {
```

Comment:
Because you want to know precise values for each of the slots I feel like.
And Since we use name properties of the components it's not hardcoded which I like.

Reply:
Let's discuss about this. This might not worth adding a place for which you have to change the code each time you add a new component in the Layout. If you want to do this dynamically, create a folder with all the layout components and list the components under this folder

Reply:
That sounds good, I'll do that!